api/v1alpha1/thanosstore_types.go

```diff
@@ -27,6 +27,10 @@ type ThanosStoreSpec struct {
 	// ObjectStorageConfig is the secret that contains the object storage configuration for Store Gateways.
 	// +kubebuilder:validation:Required
 	ObjectStorageConfig ObjectStorageConfig `json:"objectStorageConfig,omitempty"`
+	// StorageSize is the size of the storage to be used by the Thanos Store StatefulSets.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$`
+	StorageSize string `json:"storageSize"`
```

Comment:
I think we could extract and. move this into types.go and allow it to be common across components instead of repeating the validation marker?